javascript/javascript1/week2/homework4/smartEase.js

```diff
@@ -0,0 +1,29 @@
+
+function getFullnameV1(firstName, surname){
+  let fullName = `${firstName} ${surname}`
+  return fullName;
+}
+
+const fullName1 = getFullnameV1('Mark', 'Corrigan');
+const fullName2 = getFullnameV1('Jeremy', 'Usbourne');
+console.log(fullName1);
+console.log(fullName2);
+
+function getFullnameV2(firstName, surname, useFormalName=true, gender) {
+    if((useFormalName === undefined || useFormalName === false) && (gender === undefined || gender === 'other')){
```

Comment:
Nice reuse of your first implementation! 👍 If we were to assume the old one would no longer be used directly, I would probably avoid having it as a separate function, but we do not know that here. 😄

As a small nitpick, can you think of a way to avoid the `fullFormalName` and `fullName` variables? 🔍

Reply:
I did get rid of the call to v1 from inside v2, which I falsely mentioned as "nesting" in my commit message. 
I also got rid of the variables by introducing a prefix. and now it does look a lot cleaner and more concise 🙏
Can you please confirm this is the goal of your comment, otherwise I will be more than happy to tidy up more, if it is possible.

Reply:
This is a nice change! 👍 A smal thing is that maybe the gender check should be OR'd instead of AND'd. If I read this code correctly, I would expect it to yield `" Firstname Lastname"` (note the whitepace in the beginning) if you ask for a formal name but use the gender `"other"` which is a bit unexpected.

As a minor formatting thing, I would also avoid nesting ternaries as they quickly get out of hand and become hard to read. 😄

Reply:
I have fixed it. I didn't pick up on the unintended gender assignment, which is to think of it, the opposite of the initial goal. Thank you for that! 
I also for rid of the nested ternaries for a more readable if else. 
Thank you so much.